app/src/androidTest/java/org/mozilla/focus/activity/FirebaseSwitcherTest.java

```diff
@@ -0,0 +1,244 @@
+import android.support.test.runner.AndroidJUnit4;
+import android.widget.Switch;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mozilla.focus.R;
+import org.mozilla.focus.helper.ActivityRecreateLeakWatcherIdlingResource;
+import org.mozilla.focus.utils.AndroidTestUtils;
+import org.mozilla.focus.utils.FirebaseHelper;
+import org.mozilla.focus.widget.TelemetrySwitchPreference;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isChecked;
+import static android.support.test.espresso.matcher.ViewMatchers.isNotChecked;
+import static android.support.test.espresso.matcher.ViewMatchers.isRoot;
+import static android.support.test.espresso.matcher.ViewMatchers.withClassName;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.core.Is.is;
+
+// Only device with API>=24 can set default browser via system settings
+@RunWith(AndroidJUnit4.class)
+@SdkSuppress(minSdkVersion = 24, maxSdkVersion = 27)
+public class FirebaseSwitcherTest {
+
+    // idling resource for ActivityRecreateLeakWatcherIdlingResource
+    private ActivityRecreateLeakWatcherIdlingResource leakWatchIdlingResource;
+
+    @Rule
+    public final ActivityTestRule<SettingsActivity> settingsActivity = new ActivityTestRule<>(SettingsActivity.class, false, false);
+
+    @Before
+    public void setup() {
+        // make sure the pref is on when started
+        resetPref();
+
+        AndroidTestUtils.beforeTest();
+    }
+
+    @After
+    public void tearDown() {
+        if (leakWatchIdlingResource != null) {
+            IdlingRegistry.getInstance().unregister(leakWatchIdlingResource);
+        }
+    }
+
+    @Test
+    public void disableFirebase_makeSureSwitchIsOffAfterClick() {
+
+        // prepare for the view to interact
+        DataInteraction view = prepareForView();
+
+        // make sure Send Usage Data pref' switch is checked
+        view.check(matches(isChecked()));
+
+        // after click on the pref
+        view.perform(click());
+
+        // now the pref should be unchecked
+        view.check(matches(isNotChecked()));
+    }
+
+    private DataInteraction prepareForView() {
+        // Now launch Rocket's setting activity
+        settingsActivity.launchActivity(new Intent());
+
+//        FirebaseHelper.init(settingsActivity.getActivity());
+
+        FirebaseHelper.injectEnablerCallback(new Delay());
+
+        // Click on the switch multiple times...
+        return onData(
+                is(instanceOf(TelemetrySwitchPreference.class))).
```

Comment:
Can we find the preference by its key ```R.string.pref_key_default_browser``` instead of knowing the implementation class?

Reply:
TelemetrySwitchPreference's persistent is false. It has no key associated with it. I guess it's not possible to withKey() method to find it